src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -620,3 +636,250 @@ TEST_F(ControllerScriptEngineLegacyTest, connectionExecutesWithCorrectThisObject
+                     if (!(this instanceof MyClass)) {throw new Error("this should be an instance of MyClass");}
+                     engine.setValue('[Test]', 'co', this.globVar);
+                     this.globVar++;
+                     engine.setValue('[Test]', 'coTimerId', this.timerId + 10);
+                  }, true);            
+                  engine.setValue('[Test]', 'coTimerId', this.timerId);
+               }
+            }
+            var MyMapping = new MyClass();
+            MyMapping.runTimer();)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+    double timerId = coTimerId->get();
+    EXPECT_TRUE(timerId > 0);
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(timerId + 10, coTimerId->get());
+    EXPECT_DOUBLE_EQ(7.0, co->get());
+    EXPECT_TRUE(evaluateAndAssert("engine.setValue('[Test]', 'co', MyMapping.globVar);"));
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(8.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_repeatedTimerArrowFunctionCallInClass) {
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(
+            class MyClass {
+               constructor() {
+                  this.timerId = undefined;
+                  this.globVar = 7;
+               }
+               stopTimer() {
+                  if (!(this instanceof MyClass)) {throw new Error("this should be an instance of MyClass");}
+                  engine.stopTimer(this.timerId);
+                  this.timerId = 0;
+                  engine.setValue('[Test]', 'coTimerId', this.timerId + 20);
+               }
+               runTimer() {
+                  this.timerId = engine.beginTimer(20, ()=>this.stopTimer(), false);                  
+                  engine.setValue('[Test]', 'co', this.globVar);      
+                  engine.setValue('[Test]', 'coTimerId', this.timerId);
+               }
+            }
+            var MyMapping = new MyClass();
+            MyMapping.runTimer();)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(7.0, co->get());
+    double timerId = coTimerId->get();
+    EXPECT_TRUE(timerId > 0);
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(20, coTimerId->get());
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(20, coTimerId->get());
+}
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_repeatedTimerThisFunctionCallInClass) {
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(
+            class MyClass {
+               constructor() {
+                  this.timerId = undefined;
+                  this.globVar = 7;
+               }
+               stopTimer() {    
+                  if (!(this instanceof MyClass)) {throw new Error("this should be an instance of MyClass");}
```

Comment:
Small formatting nitpick.
```suggestion
                  if (!(this instanceof MyClass)) { throw new Error("this should be an instance of MyClass"); }
```